substrate/client/rpc-spec-v2/src/chain_head/chain_head_storage.rs

```diff
@@ -18,192 +18,106 @@
 
 //! Implementation of the `chainHead_storage` method.
 
-use std::{collections::VecDeque, marker::PhantomData, sync::Arc};
+use std::{marker::PhantomData, sync::Arc};
 
 use sc_client_api::{Backend, ChildInfo, StorageKey, StorageProvider};
-use sc_utils::mpsc::TracingUnboundedSender;
 use sp_runtime::traits::Block as BlockT;
+use tokio::sync::mpsc;
 
-use crate::{
-	chain_head::{
-		event::{OperationError, OperationId, OperationStorageItems},
-		subscription::BlockGuard,
-		FollowEvent,
-	},
-	common::{
-		events::{StorageQuery, StorageQueryType},
-		storage::{IterQueryType, QueryIter, QueryIterResult, Storage},
-	},
+use crate::common::{
+	events::{StorageQuery, StorageQueryType},
+	storage::{IterQueryType, QueryIter, QueryResult, Storage},
 };
 
 /// Generates the events of the `chainHead_storage` method.
 pub struct ChainHeadStorage<Client, Block, BE> {
 	/// Storage client.
 	client: Storage<Client, Block, BE>,
-	/// Queue of operations that may require pagination.
-	iter_operations: VecDeque<QueryIter>,
-	/// The maximum number of items reported by the `chainHead_storage` before
-	/// pagination is required.
-	operation_max_storage_items: usize,
 	_phandom: PhantomData<(BE, Block)>,
 }
 
+impl<Client, Block, BE> Clone for ChainHeadStorage<Client, Block, BE> {
+	fn clone(&self) -> Self {
+		Self { client: self.client.clone(), _phandom: PhantomData }
+	}
+}
+
 impl<Client, Block, BE> ChainHeadStorage<Client, Block, BE> {
 	/// Constructs a new [`ChainHeadStorage`].
-	pub fn new(client: Arc<Client>, operation_max_storage_items: usize) -> Self {
-		Self {
-			client: Storage::new(client),
-			iter_operations: VecDeque::new(),
-			operation_max_storage_items,
-			_phandom: PhantomData,
-		}
+	pub fn new(client: Arc<Client>, _operation_max_storage_items: usize) -> Self {
+		Self { client: Storage::new(client), _phandom: PhantomData }
 	}
 }
 
 impl<Client, Block, BE> ChainHeadStorage<Client, Block, BE>
 where
 	Block: BlockT + 'static,
 	BE: Backend<Block> + 'static,
-	Client: StorageProvider<Block, BE> + 'static,
+	Client: StorageProvider<Block, BE> + Send + Sync + 'static,
```

Comment:
arrgh breaking change, normally this will just be the substrate client which is already send + sync but probably a blocker for backporting this then?